src/altinn-app-frontend/src/components/base/MapComponent.test.tsx

```diff
@@ -0,0 +1,95 @@
+import * as React from 'react';
+import { render as rtlRender, screen } from '@testing-library/react';
+
+import type { IComponentProps } from 'src/components';
+import { MapComponent } from './MapComponent';
+import type { IMapComponentProps } from './MapComponent';
+
+const render = (props: Partial<IMapComponentProps> = {}) => {
+  const mockLanguage = {
+    map_component: {
+      selectedLocation: 'Selected location: {0},{1}',
+      noSelectedLocation: 'No selected location',
+    },
+  };
+
+  const allProps: IMapComponentProps = {
+    id: 'id',
+    formData: {
+      simpleBinding: undefined,
+    },
+    handleDataChange: () => '',
+    getTextResource: () => 'test',
+    isValid: true,
+    dataModelBindings: {},
+    componentValidations: {},
+    language: mockLanguage,
+    readOnly: false,
+    required: false,
+    textResourceBindings: {},
+    ...({} as IComponentProps),
+    ...props,
+  };
+
+  rtlRender(<MapComponent {...allProps} />);
+};
+
+function getButton(name: string) {
+  return screen.queryByRole('button', { name: name });
+}
+
+function getLink(name: string) {
+  return screen.queryByRole('link', { name: name });
+}
+
+function getFooterText() {
+  screen.getAllByRole('hepp');
+  return screen.queryByRole('link');
+}
+
+describe('MapComponent', () => {
+  it('should display attribution link', () => {
+    render({
+      layers: [
+        {
+          url: 'dummy',
+          attribution:
+            'Data © <a href="http://www.kartverket.no/">Kartverket</a>',
+        },
+      ],
+    });
+
+    expect(getLink('Kartverket')).toBeInTheDocument();
+  });
+
+  it('should show map with zoom buttons when readonly is false', () => {
+    render({
+      readOnly: false,
+    });
+
+    expect(getButton('Zoom in')).toBeInTheDocument();
+    expect(getButton('Zoom out')).toBeInTheDocument();
+  });
+
+  it('should show map without zoom buttons when readonly is true', () => {
+    render({
+      readOnly: true,
+    });
+
+    expect(getButton('Zoom in')).not.toBeInTheDocument();
+    expect(getButton('Zoom out')).not.toBeInTheDocument();
+  });
```

Comment:
These tests probably makes more sense to move to the tests in altinn-design-system repo instead?